ti/instrumentation/utils.go

```diff
@@ -264,6 +267,151 @@ func getChangedFiles(ctx context.Context, workspace string, log *logrus.Logger,
 	return res, nil
 }
 
+// addBazelFilesToChangedFiles takes a list of files and removes bazel build files and adds java files listed in target src globs
+func addBazelFilesToChangedFiles(ctx context.Context, workspace string, log *logrus.Logger, res []ti.File, bazelFileCount string) ([]ti.File, []string, error) {
+	var moduleList []string
+	threshold, err := strconv.Atoi(bazelFileCount)
+	if err != nil {
+		return res, moduleList, fmt.Errorf("bazelFileCount not set correctly in ticonfig.yml, expecting number found character %v ", bazelFileCount)
+	}
+
+	// map to prevent duplicate files
+	uniqueFiles := make(map[string]struct{})
+
+	var changedRes []ti.File
+	for _, file := range res {
+		if !strings.HasSuffix(file.Name, "BUILD.bazel") {
```

Comment:
Bazel convention is `BUILD` but Harness seems to use `BUILD.bazel` so please add a comment here that this needs to change when we support this feature more generically for bazel. Also make sure we are not using `BUILD` anywhere in harness-core